tls/extensions/s2n_client_key_share.c

```diff
@@ -98,6 +95,62 @@ static int s2n_ecdhe_supported_curves_send(struct s2n_connection *conn, struct s
     return S2N_SUCCESS;
 }
 
+static int s2n_send_hrr_keyshare(struct s2n_connection *conn, struct s2n_stuffer *out)
+{
+    notnull_check(conn);
+    const struct s2n_ecc_named_curve *server_negotiated_curve = NULL;
+    struct s2n_ecc_evp_params *ecc_evp_params = NULL;
+
+    const struct s2n_ecc_preferences *ecc_pref = NULL;
+    GUARD(s2n_connection_get_ecc_preferences(conn, &ecc_pref));
+    notnull_check(ecc_pref);
+
+    server_negotiated_curve = conn->secure.server_ecc_evp_params.negotiated_curve;
+    S2N_ERROR_IF(server_negotiated_curve == NULL, S2N_ERR_BAD_KEY_SHARE);
```

Comment:
It's probably best not to mix `S2N_ERROR_IF` and `ENSURE_POSIX` in the same function

Reply:
Agreed, adopting the `ENSURE_POSIX` :D